hms-app/src/components/AllHackathonList.tsx

```diff
@@ -0,0 +1,102 @@
+import { Group, Button, createStyles, Card, SimpleGrid } from '@mantine/core'
+import React, { useEffect, useState } from 'react'
+import { getListOfHackathons } from '../actions/HackathonActions'
+import { HackathonPreview } from '../common/types'
+
+const useStyles = createStyles((theme) => ({
+  card: {
+    backgroundColor:
+      theme.colorScheme === 'dark' ? theme.colors.dark[7] : theme.white,
+  },
+  list: {
+    backgroundColor:
+      theme.colorScheme === 'dark' ? theme.colors.dark[7] : theme.white,
+  },
+  list2: {
+    backgroundColor:
+      theme.colorScheme === 'dark'
+        ? theme.colors.dark[4]
+        : theme.colors.gray[3],
+  },
+  section: {
+    borderBottom: `1px solid ${
+      theme.colorScheme === 'dark' ? theme.colors.dark[4] : theme.colors.gray[3]
+    }`,
+    paddingLeft: theme.spacing.md,
+    paddingRight: theme.spacing.md,
+    paddingBottom: theme.spacing.md,
+    /*textTransform: 'uppercase',*/
+    fontSize: theme.fontSizes.md,
+    fontWeight: 500,
+  },
+}))
+
+function AllHackathonList() {
+  const { classes } = useStyles()
+  const [hackathonList, setHackathonList] = useState({
+    errorHackathonList: false,
+    isLoadingHackathonList: true,
+    hackathons: [] as HackathonPreview[],
+  })
```

Comment:
so the anal-retentive part of me would say that this is more than just the `hackathonList`.  I can't remember completely but I think in my React class they said things like this would generally get split apart.  So you would have something more like;
```
const [isError, setIsError] = useState(false);
const [isLoading, setIsLoading] = useState(false);
const [hackathonsList, setHackathonsList] = useState([] as HackathonPreview[]);
```

then it becomes a bit easier if you really only need to change one of these things when something happens and you need not do so much `...hackathonList` stuff.  But it isn't anything to change right now.  Maybe play with it a bit and see which you like more or is easier to understand when reading the code.